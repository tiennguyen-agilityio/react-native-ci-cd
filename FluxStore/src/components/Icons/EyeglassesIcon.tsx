import {memo} from 'react';
import {TouchableOpacity} from 'react-native';
import Svg, {Path} from 'react-native-svg';

import {IconProps} from '@/interfaces';
import {useThemeStore} from '@/stores';

export const EyeglassesIcon = memo(
  ({
    width = 21,
    height = 15,
    color,
    disabled = false,
    isActive = false,
    style,
    onPress,
  }: IconProps) => {
    const {
      theme: {icon, text},
    } = useThemeStore();

    return (
      <TouchableOpacity
        testID="eyeglasses-icon"
        activeOpacity={0.8}
        onPress={onPress}
        disabled={disabled}
        style={style}>
        <Svg width={width} height={height} viewBox="0 0 21 15" fill="none">
          <Path
            d="M6.742 0a2.6 2.6 0 00-1.967.9L.47 5.88A1.942 1.942 0 000 7.15v3.9a3.25 3.25 0 003.25 3.25h2.6a3.25 3.25 0 003.25-3.25V9.1h2.6v1.95a3.25 3.25 0 003.25 3.25h2.6a3.25 3.25 0 003.25-3.25v-3.9c0-.558-.234-1.06-.608-1.414L16.025.902A2.6 2.6 0 0014.057 0H13a.65.65 0 100 1.3h1.057a1.3 1.3 0 01.984.451L18.014 5.2H13.65a1.95 1.95 0 00-1.95 1.95v.65H9.1v-.65A1.95 1.95 0 007.15 5.2H2.775L5.76 1.75a1.3 1.3 0 01.983-.45H7.8a.65.65 0 100-1.3H6.742zM7.8 7.15v3.9A1.95 1.95 0 015.85 13h-2.6a1.95 1.95 0 01-1.95-1.95v-3.9a.65.65 0 01.65-.65h5.2a.65.65 0 01.65.65zm5.2 3.9v-3.9a.65.65 0 01.65-.65h5.2c.156 0 .299.055.41.147l.072.082.008-.005c.103.118.16.27.16.426v3.9A1.95 1.95 0 0117.55 13h-2.6A1.95 1.95 0 0113 11.05z"
            fill={color ?? isActive ? text.light : icon.tertiary}
          />
        </Svg>
      </TouchableOpacity>
    );
  },
);
