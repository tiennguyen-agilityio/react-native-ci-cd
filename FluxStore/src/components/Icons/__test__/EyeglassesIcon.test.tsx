import {render, fireEvent, screen} from '@testing-library/react-native';

import {EyeglassesIcon} from '../EyeglassesIcon';

describe('EyeglassesIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<EyeglassesIcon />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with isActive', () => {
    const {toJSON} = render(<EyeglassesIcon isActive />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when touched and not disabled', () => {
    render(<EyeglassesIcon onPress={onPress} />);

    fireEvent.press(screen.getByTestId('eyeglasses-icon'));

    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does NOT call onPress when disabled', () => {
    render(<EyeglassesIcon onPress={onPress} disabled />);

    fireEvent.press(screen.getByTestId('eyeglasses-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });
});
