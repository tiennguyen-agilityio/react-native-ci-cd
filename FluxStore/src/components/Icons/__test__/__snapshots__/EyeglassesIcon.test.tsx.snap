// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EyeglassesIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="eyeglasses-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={15}
    bbWidth={21}
    fill="none"
    focusable={false}
    height={15}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 15,
          "width": 21,
        },
      ]
    }
    vbHeight={15}
    vbWidth={21}
    width={21}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M6.742 0a2.6 2.6 0 00-1.967.9L.47 5.88A1.942 1.942 0 000 7.15v3.9a3.25 3.25 0 003.25 3.25h2.6a3.25 3.25 0 003.25-3.25V9.1h2.6v1.95a3.25 3.25 0 003.25 3.25h2.6a3.25 3.25 0 003.25-3.25v-3.9c0-.558-.234-1.06-.608-1.414L16.025.902A2.6 2.6 0 0014.057 0H13a.65.65 0 100 1.3h1.057a1.3 1.3 0 01.984.451L18.014 5.2H13.65a1.95 1.95 0 00-1.95 1.95v.65H9.1v-.65A1.95 1.95 0 007.15 5.2H2.775L5.76 1.75a1.3 1.3 0 01.983-.45H7.8a.65.65 0 100-1.3H6.742zM7.8 7.15v3.9A1.95 1.95 0 015.85 13h-2.6a1.95 1.95 0 01-1.95-1.95v-3.9a.65.65 0 01.65-.65h5.2a.65.65 0 01.65.65zm5.2 3.9v-3.9a.65.65 0 01.65-.65h5.2c.156 0 .299.055.41.147l.072.082.008-.005c.103.118.16.27.16.426v3.9A1.95 1.95 0 0117.55 13h-2.6A1.95 1.95 0 0113 11.05z"
        fill={
          {
            "payload": 4288519581,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`EyeglassesIcon renders correctly (matches snapshot) with isActive 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="eyeglasses-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={15}
    bbWidth={21}
    fill="none"
    focusable={false}
    height={15}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 15,
          "width": 21,
        },
      ]
    }
    vbHeight={15}
    vbWidth={21}
    width={21}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M6.742 0a2.6 2.6 0 00-1.967.9L.47 5.88A1.942 1.942 0 000 7.15v3.9a3.25 3.25 0 003.25 3.25h2.6a3.25 3.25 0 003.25-3.25V9.1h2.6v1.95a3.25 3.25 0 003.25 3.25h2.6a3.25 3.25 0 003.25-3.25v-3.9c0-.558-.234-1.06-.608-1.414L16.025.902A2.6 2.6 0 0014.057 0H13a.65.65 0 100 1.3h1.057a1.3 1.3 0 01.984.451L18.014 5.2H13.65a1.95 1.95 0 00-1.95 1.95v.65H9.1v-.65A1.95 1.95 0 007.15 5.2H2.775L5.76 1.75a1.3 1.3 0 01.983-.45H7.8a.65.65 0 100-1.3H6.742zM7.8 7.15v3.9A1.95 1.95 0 015.85 13h-2.6a1.95 1.95 0 01-1.95-1.95v-3.9a.65.65 0 01.65-.65h5.2a.65.65 0 01.65.65zm5.2 3.9v-3.9a.65.65 0 01.65-.65h5.2c.156 0 .299.055.41.147l.072.082.008-.005c.103.118.16.27.16.426v3.9A1.95 1.95 0 0117.55 13h-2.6A1.95 1.95 0 0113 11.05z"
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
